pkg/devfile/devfile.go

```diff
@@ -341,20 +341,40 @@ func GetResourceFromDevfile(log logr.Logger, devfileData data.DevfileData, deplo
 
 					if currentPort > 0 {
 						servicePort := corev1.ServicePort{
+							Name:       strconv.Itoa(currentPort),
 							Port:       int32(currentPort),
 							TargetPort: intstr.FromInt(currentPort),
 						}
 
 						isPresent := false
+						portNameMap := make(map[string]bool)
 						for _, port := range resources.Services[0].Spec.Ports {
+							portNameMap[port.Name] = true
 							if port.Port == servicePort.Port {
 								isPresent = true
-								break
 							}
 						}
 
 						if !isPresent {
+							if portNameMap[servicePort.Name] {
+								generatedName := fmt.Sprintf("%s-%s", servicePort.Name, util.GetRandomString(4, true))
+								portNameMap[generatedName] = true
+								servicePort.Name = generatedName
+							}
 							resources.Services[0].Spec.Ports = append(resources.Services[0].Spec.Ports, servicePort)
+
+							for i, port := range resources.Services[0].Spec.Ports {
```

Comment:
any reason we are iterating through the ports twice?

Reply:
sounds reasonable to me, 
just one additional thing,  since the first loop is just to check the existence. if the port ever present, we should break out of the loop.

Reply:
We dont break because I need to populate `portNameMap[port.Name] = true` on first loop

Then use the map to see if our Component spec port name needs to be updated, when appending